File: Examples/Features/Workbook/ImportCsvFileToWorkbookWithOptions.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Workbook
{
    public class ImportCsvFileToWorkbookWithOptions : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            //Open csv with custom delimiter.
            CsvOpenOptions options = new CsvOpenOptions
            {
                ColumnSeparator = ",",
                RowSeparator = "\r\n",
                CellSeparator = '"'
            };

            //Change the path to the real file path when open.
            workbook.Open(System.IO.Path.Combine(this.CurrentDirectory, "source.csv"), options);
        }

        public override bool CanDownload
        {
            get
            {
                return false;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }
    }

}

File: Examples.VB/Features/Workbook/ImportCsvFileToWorkbookWithOptions.vb
Namespace Features.Workbook
    Public Class ImportCsvFileToWorkbookWithOptions
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Open csv with custom delimiter.
            Dim options As New CsvOpenOptions With {
                .ColumnSeparator = ",",
                .RowSeparator = vbCrLf,
                .CellSeparator = """"c
            }

            'Change the path to the real file path when open.
            workbook.Open(IO.Path.Combine(CurrentDirectory, "source.csv"), options)
        End Sub
        Public Overrides ReadOnly Property CanDownload As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
